FF136 Relnote: Cookie Store API (#38096)
mdn/content: files/en-us/mozilla/firefox/releases/136/index.md

### APIs

- The maximum size of [Data URLs](/en-US/docs/Web/URI/Schemes/data) has been increased from 32MB to 512MB, matching the limit for Chromium browsers ([Firefox bug 1911300](https://bugzil.la/1911300)).

> - The [Cookie Store API](/en-US/docs/Web/API/Cookie_Store_API) is a modern, {{glossary("asynchronous")}} {{jsxref("Promise")}}-based method of managing cookies, which can be used in both the main thread and in [service workers](/en-US/docs/Web/API/Service_Worker_API).
> 
>   A subset of the Cookie Store API has been implemented ([Firefox bug 1800882](https://bugzil.la/1800882), [Firefox bug 1937477](https://bugzil.la/1937477)). This includes:

Reviewer: Based on the preview, the newlines are swallowed up, leave it as-is if you like it:

```suggestion
- The [Cookie Store API](/en-US/docs/Web/API/Cookie_Store_API) is a modern, {{glossary("asynchronous")}} {{jsxref("Promise")}}-based method of managing cookies, which can be used in both the main thread and in [service workers](/en-US/docs/Web/API/Service_Worker_API).
  A subset of the Cookie Store API has been implemented ([Firefox bug 1800882](https://bugzil.la/1800882), [Firefox bug 1937477](https://bugzil.la/1937477)). This includes:
```

Reviewer: https://bugzil.la/1800882 is:

> Implement HTMLVideoElement.requestVideoFrameCallback()

This is probably just a copy-paste err from the experimental features page, we can likely omit it entirely

Author: Good catch. I think it might have been - I "trusted" the expr features content. Modified your change above to remove it.

Yes, I guess we'll swallow the new lines, though this is IMO a flaw in the rendering of bullets.

  - The [`CookieStore`](/en-US/docs/Web/API/CookieStore) interface, for getting, setting, and deleting cookies.
  - The [`Window.cookieStore`](/en-US/docs/Web/API/Window/cookieStore) and [`ServiceWorkerGlobalScope.cookieStore`](/en-US/docs/Web/API/ServiceWorkerGlobalScope/cookieStore) properties for getting `CookieStore` instances.

>   - The [`changed` event](/en-US/docs/Web/API/CookieChangeEvent/changed) and [`deleted` event](/en-US/docs/Web/API/CookieChangeEvent/deleted) (and their interface [`CookieChangeEvent`](/en-US/docs/Web/API/CookieChangeEvent)), which can be fired in either main thread or service worker contexts.

Author: ```suggestion
  - The [`change` event](/en-US/docs/Web/API/CookieStore/change_event) (and its interface [`CookieChangeEvent`](/en-US/docs/Web/API/CookieChangeEvent)), which fires in main thread and service worker contexts when cookies are set or deleted.
```

>   Note that while any of the supported cookie properties can be [set](/en-US/docs/Web/API/CookieStore/get), the cookie objects returned by the [`get()`](/en-US/docs/Web/API/CookieStore/get) and [`getAll()`](/en-US/docs/Web/API/CookieStore/getAll) methods, and in events, omit all properties other than `name` and `value` (matching the information returned by the {{domxref("document.cookie")}}).

Author: ```suggestion
  Note that while any of the supported cookie properties can be [set](/en-US/docs/Web/API/CookieStore/get), the cookie objects returned by the [`get()`](/en-US/docs/Web/API/CookieStore/get) and [`getAll()`](/en-US/docs/Web/API/CookieStore/getAll) methods, and in the `change` event, omit all properties other than `name` and `value` (matching the information returned by the {{domxref("document.cookie")}}).
```

>   Note that while any of the supported cookie properties can be [set](/en-US/docs/Web/API/CookieStore/get), the cookie objects returned by the [`get()`](/en-US/docs/Web/API/CookieStore/get) and [`getAll()`](/en-US/docs/Web/API/CookieStore/getAll) methods, and in the `change` event, omit all properties other than `name` and `value` (matching the information returned by the {{domxref("document.cookie")}}).
> 
>   The following interfaces and properties are not implemented: [`ServiceWorkerRegistration.cookies`](/en-US/docs/Web/API/ServiceWorkerRegistration/cookies), [`CookieStoreManager`](/en-US/docs/Web/API/CookieStoreManager), and [`ExtendableCookieChangeEvent`](/en-US/docs/Web/API/ExtendableCookieChangeEvent).

Reviewer: Same comment:

```suggestion
  Note that while any of the supported cookie properties can be [set](/en-US/docs/Web/API/CookieStore/get), the cookie objects returned by the [`get()`](/en-US/docs/Web/API/CookieStore/get) and [`getAll()`](/en-US/docs/Web/API/CookieStore/getAll) methods, and in the `change` event, omit all properties other than `name` and `value` (matching the information returned by the {{domxref("document.cookie")}}).
  The following interfaces and properties are not implemented: [`ServiceWorkerRegistration.cookies`](/en-US/docs/Web/API/ServiceWorkerRegistration/cookies), [`CookieStoreManager`](/en-US/docs/Web/API/CookieStoreManager), and [`ExtendableCookieChangeEvent`](/en-US/docs/Web/API/ExtendableCookieChangeEvent).
```
